compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -8075,7 +8075,10 @@ public void visit(BLangTypeConversionExpr conversionExpr) {
 
     @Override
     public void visit(BLangLambdaFunction bLangLambdaFunction) {
+        BLangOnFailClause onFailClause = this.onFailClause;
+        this.onFailClause = null;
```

Comment:
It would be better to move this logic to the visitor of the `blangfunction`. Also, @mindula, is this the only global variable we need to reset related to the `onfail` clause?